### ISSUE TITLE
MaterialDatePicker background drawn on top of header background

### ISSUE THREAD

**Issue by Snezhe** [material-components/material-components-android#2588]
Description:
I'm using Material Components I am trying to apply a custom style to a MaterialDatePicker.
I'm trying to set a different background color for the body and for the header. While the header technically uses the correct background color (from the materialCalendarHeaderLayout style), the dialog background still gets partially drawn on top.

Expected behavior:
The header and body should have separate background colors

Actual behavior:

![Screenshot_1646298061](https://user-images.githubusercontent.com/46113857/156532204-7c5434c6-31e2-47f4-9586-0a0626e81364.png)

This is my source code in styles.xml:

    <style name="materialRangeDatePicker" parent="@style/ThemeOverlay.MaterialComponents.MaterialCalendar">
        <item name="colorPrimary">?attr/colorD</item>
        <item name="colorOnPrimary">?attr/colorG</item>
        <item name="materialCalendarHeaderLayout">@style/materialCalendarHeaderLayout</item>
        <item name="materialCalendarHeaderTitle">@style/materialCalendarHeaderTitle</item>
        <item name="materialCalendarStyle">@style/myMaterialCalendar</item>
        <item name="shapeAppearanceSmallComponent">@style/ShapeAppearance.App.SmallComponent</item>
        <item name="android:background">?attr/colorA</item>
    </style>

    <style name="myMaterialCalendar" parent="Widget.MaterialComponents.MaterialCalendar">
        <item name="rangeFillColor">?attr/colorE</item>
        <item name="android:windowFullscreen">true</item>
        <item name="dayTodayStyle">@style/materialCalendarDayTodayStyle</item>
    </style>

    <style name="materialCalendarDayTodayStyle" parent="Widget.MaterialComponents.MaterialCalendar">
        <item name="itemFillColor">?attr/colorC</item>
        <item name="itemTextColor">?attr/colorG</item>
        <item name="itemStrokeWidth">0dp</item>
        <item name="itemShapeAppearance">
            @style/Test.ShapeAppearanceOverlay.MaterialComponents.MaterialCalendar.Day
        </item>
    </style>

    <style name="materialCalendarHeaderLayout" parent="Widget.MaterialComponents.MaterialCalendar.HeaderLayout">
        <item name="android:background">?attr/colorB</item>
    </style>

    <style name="materialCalendarHeaderTitle" parent="Widget.MaterialComponents.MaterialCalendar.HeaderTitle">
        <item name="android:text">@string/select_period</item>
        <item name="android:textColor">?attr/colorG</item>
        <item name="android:textSize">20sp</item>
    </style>
    
    <style name="ShapeAppearance.App.SmallComponent" parent="ShapeAppearance.MaterialComponents.SmallComponent">
        <item name="android:textSize">24sp</item>
        <item name="cornerSize">16dp</item>
    </style>

And I am showing the dialog with: 

                    val dateRangePicker = MaterialDatePicker.Builder.dateRangePicker()
                            .setTitleText("Select dates")
                            .setSelection(
                                    Pair(
                                            MaterialDatePicker.thisMonthInUtcMilliseconds(),
                                            MaterialDatePicker.todayInUtcMilliseconds()
                                    )
                            )
                            .setTheme(R.style.materialRangeDatePicker)
                            .build()
                    activity?.let { dateRangePicker.show(activity.supportFragmentManager, "date_range_picker") }

**Comment by Snezhe** [comment]
@drchen

**Comment by drchen** [comment]
I think you need to move `<item name="android:background">?attr/colorA</item>` from `materialRangeDatePicker` to `myMaterialCalendar` instead. The theme settings will be applied to all views which may cause the problem I think.

Let me know if this solves your issue.

**Comment by Snezhe** [comment]
@drchen Yes this does solve the background color on top of the header, thank you :) 
But now my background color is transparent. Do you have any idea what might cause this?

![Screenshot_1646325032](https://user-images.githubusercontent.com/46113857/156608237-7230ec05-86f0-4b98-bcd5-b1b0310e5f80.png)

**Comment by drchen** [comment]
Hmm... I'm not entirely sure. Are you overriding ?attr/colorSurface anywhere?

**Comment by Snezhe** [comment]
@drchen  No i'm not

**Comment by drchen** [comment]
That's very weird. Can you let me know what colorA and colorB is?

**Comment by Snezhe** [comment]
@drchen colorA is : #343434;
colorB is: #484848

**Comment by Snezhe** [comment]
@drchen   Okay so I set the surfaceColor instead of background color in materialRangeDatePicker and it seems to be working now
